presentation/components/src/presentation-components/table/DataProvider.ts

```diff
@@ -256,9 +257,57 @@ const createColumns = (descriptor: Readonly<Descriptor> | undefined): ColumnDesc
     return [createLabelColumn()];
 
   const sortedFields = [...descriptor.fields].sort(priorityAndNameSortFunction);
+
+  extractSameInstanceFields(sortedFields);
+
   return sortedFields.map((field) => createColumn(field));
 };
 
+const extractSameInstanceFields = (fields: Field[]) => {
+  const foundFields: { [field: string]: Field } = {};
+  for (let i = 0; i < fields.length; i++) {

```

Comment:
As far as I can tell this won't handle the case when there's a "same instance" nested content field `A` nested inside another similar field `B` - we're going to replace `B` with `A` and increase the index, which means we're not going to handle `A`.

Reply:
Updated